src/main/frontend/components/ReviewTile.js

```diff
@@ -12,13 +19,80 @@ const ReviewTile = (props) => {
     userReview = "User Review: "
   }
 
+  const voted = count => {
+    let localCount = stateCount
+    if (downvoted == true && count == 1 && upvoted == false) {
+      setDownvoted(false)
+      setUpvoted(true)
+      localCount += 2
+    } else if (upvoted == true && count == -1 && downvoted == false) {
+      setUpvoted(false)
+      setDownvoted(true)
+      localCount -= 2
+    } else if (count == 1 && upvoted == true) {
+      setUpvoted(false)
+      setDownvoted(false)
+      localCount -= 1
+    } else if (count == 1 && upvoted == false) {
+      setUpvoted(true)
+      setDownvoted(false)
+      localCount += 1
+    } else if (count == -1 && downvoted == true) {
+      setDownvoted(false)
+      setUpvoted(false)
+      localCount += 1
+    } else if (count == -1 && downvoted == false) {
+      setDownvoted(true)
+      setUpvoted(false)
+      localCount -= 1
+    }
+    setCount(localCount)
+    updateVoteCount(id, localCount)
+  }

```

Comment:
Hmmm. Good work with some tricky logic here!

A recommendation: you could get this working by simply using `-1`, `0`, and `1` to indicate the status of the vote, rather than storing boolean `upvoted` and `downvoted` attributes.

So in state, you'd store a variable called `votedStatus` which would be equal to `-1` if they'd downvoted, `0` if there's no vote, and `1` if they upvoted.

Then, you can update your conditional above to just check the value of that `votedStatus` and the `count` and go from there.